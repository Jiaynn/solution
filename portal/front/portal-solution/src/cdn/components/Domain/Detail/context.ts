import React from 'react'

import { IDomainDetail } from 'cdn/apis/domain'

export interface DomainDetailCtxValue {
  /** 刷新域名信息 */
  refreshDomainDetail: () => void
  /** 域名信息 */
  domainDetail: IDomainDetail
}

/** 域名详情上下文 */
export const domainDetailCtx = React.createContext<DomainDetailCtxValue>({
  refreshDomainDetail: () => null,
  domainDetail: null!
})

/** 用于获取域名详情的上下文信息 */
export function useDomainDetailCtx() {
  return React.useContext(domainDetailCtx)
}
