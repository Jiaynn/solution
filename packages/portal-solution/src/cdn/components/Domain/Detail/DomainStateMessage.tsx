import React from 'react'

import { Link } from 'portal-base/common/router'

import { shouldForbidConfiguration } from 'cdn/transforms/domain'
import { getMessageForOperationFailure } from 'cdn/transforms/domain/messages'

import { OperatingState, DomainType } from 'cdn/constants/domain'

import DomainStateProcessingMessage from './DomainStateProcessingMessage'
import { useDomainDetailCtx } from './context'

export default function DomainStateMessage() {
  const { domainDetail } = useDomainDetailCtx()
  const { operationType, operatingState, operatingStateDesc } = domainDetail

  switch (operatingState) {
    case OperatingState.Failed:
      return (
        <p className="error-message">
          {getMessageForOperationFailure(operationType, operatingStateDesc)}
        </p>
      )
    case OperatingState.Processing: {
      return <DomainStateProcessingMessage />
    }
    case OperatingState.Frozen:
      return (
        <p className="warning-message">该域名已被冻结，如有疑问请创建工单联系我们</p>
      )
    case OperatingState.Offlined:
      return (
        <p className="warning-message">该域名已停用，不支持自助修改配置</p>
      )
    default:
      if (!shouldForbidConfiguration(domainDetail)) {
        return null
      }
      return domainDetail.type !== DomainType.Test
        ? <p className="warning-message">该域名暂时不支持自助修改配置，请创建工单联系我们进行人工修改</p>
        : <p className="warning-message">测试域名无法自助修改配置，建议<Link to="create">创建自定义域名</Link></p>
  }
}
