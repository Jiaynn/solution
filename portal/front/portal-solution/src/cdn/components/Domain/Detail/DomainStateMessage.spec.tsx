/*
 * @file cases for domain create component
 * @author nighca <nighca@live.cn>
 */

import React from 'react'

import { RendererUtils as Renderer } from 'test'

import mockDomainDetail from 'cdn/test/domain-detail-mock'

import {
  OperatingState, operatingStateDescs, WorkflowTaskType,
  WorkflowTaskErrorCode, OperationType
} from 'cdn/constants/domain'

import DomainStateMessage from './DomainStateMessage'
import { domainDetailCtx } from './context'

const domain = mockDomainDetail()

it('renders correctly', () => {
  const renderer = new Renderer()
  const ctxValue = { domainDetail: domain, refreshDomainDetail: () => null }
  const tree = renderer.createWithAct(
    <domainDetailCtx.Provider value={ctxValue}>
      <DomainStateMessage />
    </domainDetailCtx.Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with https conf processing', () => {
  const domainDetail = {
    ...domain,
    operationType: OperationType.ModifyHttpsConf,
    operatingState: OperatingState.Processing
  }
  const ctxValue = { domainDetail, refreshDomainDetail: () => null }
  const renderer = new Renderer()
  const tree = renderer.createWithAct(
    <domainDetailCtx.Provider value={ctxValue}>
      <DomainStateMessage />
    </domainDetailCtx.Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with https cert process failed', () => {
  const domainDetail = {
    ...domain,
    operationType: OperationType.ModifyHttpsCert,
    operatingState: OperatingState.Failed,
    operatingStateDesc: operatingStateDescs.verifyHttpsCertFail
  }
  const ctxValue = { domainDetail, refreshDomainDetail: () => null }
  const renderer = new Renderer()

  const tree = renderer.createWithAct(
    <domainDetailCtx.Provider value={ctxValue}>
      <DomainStateMessage />
    </domainDetailCtx.Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with redo workflow task', () => {
  const domainDetail = {
    ...domain,
    operatingState: OperatingState.Processing,
    operTaskId: 'Task6041d314f9537d4e57000004',
    operTaskType: WorkflowTaskType.All,
    operTaskErrCode: WorkflowTaskErrorCode.QpsLimit
  }
  const ctxValue = { domainDetail, refreshDomainDetail: () => null }
  const renderer = new Renderer()

  const tree = renderer.createWithAct(
    <domainDetailCtx.Provider value={ctxValue}>
      <DomainStateMessage />
    </domainDetailCtx.Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with abandon workflow task', () => {
  const domainDetail = {
    ...domain,
    operatingState: OperatingState.Processing,
    operTaskId: 'Task6041d314f9537d4e57000004',
    operTaskType: WorkflowTaskType.Abandon,
    operTaskErrCode: WorkflowTaskErrorCode.FreeCertLimit
  }
  const ctxValue = { domainDetail, refreshDomainDetail: () => null }
  const renderer = new Renderer()

  const tree = renderer.createWithAct(
    <domainDetailCtx.Provider value={ctxValue}>
      <DomainStateMessage />
    </domainDetailCtx.Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
