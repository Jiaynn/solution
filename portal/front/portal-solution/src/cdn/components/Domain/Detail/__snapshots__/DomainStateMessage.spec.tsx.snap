// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `null`;

exports[`renders correctly with abandon workflow task 1`] = `
<p
  className="warning-message"
  style={
    Object {
      "maxWidth": "100%",
    }
  }
>
  由于同一主域名绑定的免费 DV 证书不能超过 20 个，目前额度已达上限，处理无法继续，您可以选择以下处理方案：
  <br />
  1. 点击 
  <a
    onClick={[Function]}
  >
    回滚
  </a>
  ，成功后域名即恢复可操作状态。
  <br />
  2. 可 
  <a
    className="comp-router-link"
    href="/certificate/apply"
    onClick={[Function]}
    target="_blank"
  >
    购买 DV 通配符证书
  </a>
  ，或使用新的主域名后再绑定免费证书。
  <br />
  3. 配置好证书后，重新提交域名配置修改。
</p>
`;

exports[`renders correctly with https cert process failed 1`] = `
<p
  className="error-message"
>
  HTTPS 证书有误
</p>
`;

exports[`renders correctly with https conf processing 1`] = `
<p
  className="warning-message"
>
  修改 HTTPS 配置处理中，通常情况下 8-15 分钟完成，期间域名访问不受影响，部分配置不可修改，若较长时间未配置完成，请提交工单反馈
</p>
`;

exports[`renders correctly with redo workflow task 1`] = `
<p
  className="warning-message"
>
  由于处理 QPS 限制，处理无法继续，您可以选择以下处理方案：
  <br />
  1. 点击 
  <a
    onClick={[Function]}
  >
    重试
  </a>
   尝试重新提交任务。
  <br />
  2. 如果失败，可稍后再尝试，多次尝试后仍然失败，可 
  <a
    href="https://support.qiniu.com/tickets/new/form?category=%E9%85%8D%E7%BD%AE%E9%97%AE%E9%A2%98&space=CDN"
    rel="noopener"
    target="_blank"
  >
    联系技术支持
  </a>
   处理。
</p>
`;
